Dockerfile

```diff
@@ -1,31 +1,45 @@
-FROM quay.io/opendatahub/text-generation-inference:stable-bafd218
-
-USER root
+FROM registry.access.redhat.com/ubi8/ubi-minimal:latest as base
 
 # Add grpc-ecosystem health probe
 ARG GRPC_HEALTH_PROBE_VERSION=v0.4.20
 
-WORKDIR /caikit
-COPY caikit /caikit
-
-RUN yum -y update && yum -y install git git-lfs && yum clean all && \
-    git lfs install && \
-    pip install 'micropipenv[toml]' && \
-    micropipenv install && \
-    rm -rf ~/.cache && \
-    mkdir -p /opt/models && \
-    adduser -g 0 -u 1001 caikit --home-dir /caikit && \
-    chown -R 1001:0 /caikit /opt/models && \
-    chmod -R g=u /caikit /opt/models
-
 # This is for the use-cases without kserve
 RUN curl -Lo /usr/local/bin/grpc_health_probe https://github.com/grpc-ecosystem/grpc-health-probe/releases/download/${GRPC_HEALTH_PROBE_VERSION}/grpc_health_probe-linux-amd64 && \
     chmod +x /usr/local/bin/grpc_health_probe
 
-USER 1001
+RUN microdnf -y update && \
+    microdnf -y install \
+        shadow-utils git git-lfs python39-pip python39-wheel && \
+    git lfs install && \
+    pip3 install --no-cache-dir --upgrade pip wheel && \
+    microdnf clean all
+
+
+FROM base as poetry-builder
+ENV POETRY_VIRTUALENVS_IN_PROJECT=1
+
+WORKDIR /tmp/poetry
+COPY pyproject.toml .
+COPY poetry.lock .
+RUN pip3 install poetry && poetry install
+
+
+FROM base as deploy
```

Comment:
Oh, actually upon reading deeper here, yeah, we could sidestep holding some of the holding onto stuff like devel pkgs by changing this to be the minimal container, rather than chaining on top of base

This is also good for cleaning out things like the poetry lock from the shipping image

Reply:
So when we do from base as deploy, I may be wrong, but I believe it keeps everything from the previous layers that were created, so we'd have things like `poetry.lock` in the end container. In general, best practice is probably not to have that type of file in the final build, and we're already copying --from poetry-builder, so we could instead do ubi-minimal as deploy, and whatever you want as base since we copy the .venv anyway

Reply:
We do need python in the deploy stage, and that is why I'm using `FROM base as deploy`, I doubt `git-lfs` and the grpc endpoints are actually required, but they were present initially

Ideally, the deploy stage could be really minimal:
```dockerfile
FROM registry.access.redhat.com/ubi8/ubi-minimal:latest as deploy
WORKDIR /caikit

RUN microdnf -y update && \
    microdnf -y install shadow-utils python39 && \
    microdnf clean all && \
    groupadd --system caikit --gid 1001 && \
    adduser --system --uid 1001 --gid 0 --groups caikit \
    --create-home --home-dir /caikit --shell /sbin/nologin \
    --comment "Caikit User" caikit

COPY --from=poetry-builder /tmp/poetry/.venv /caikit/

ENV VIRTUAL_ENV=/caikit
ENV PATH="$VIRTUAL_ENV/bin:$PATH"

USER caikit

ENV CONFIG_FILES=/caikit/config/caikit.yml
VOLUME ["/caikit/config/"]

CMD ["python",  "-m", "caikit.runtime.grpc_server"]
```